src/modules/components/wallet/wallet.tsx

```diff
@@ -0,0 +1,99 @@
+import classNames from 'classnames';
+import { useAccount, useConnect, useDisconnect, useEnsName, type CreateConnectorFn } from 'wagmi';
+import { injected, walletConnect, type InjectedParameters } from 'wagmi/connectors';
+import { StateSkeletonBar, StateSkeletonCircular } from '../../../core';
+import { addressUtils } from '../../utils';
+import { MemberAvatar } from '../member';
+
+export interface IWalletProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+    /**
+     * Callback fired when the user connects their wallet.
+     */
+    onConnect?: () => void;
+    /**
+     * Callback fired when the user disconnects their wallet.
+     */
+    onDisconnect?: () => void;
```

Comment:
Given that we'll probably have a more complex handling for connecting / disconnecting the user on the app (display custom dialog for connect and display user dialog when user is connected), I wouldn't add here the logic of actually connecting and disconnecting the user.
I'm also not sure if we should couple this component to the connected user state on wagmi, I would probably just let the parent component pass the information about the connected user, we could use the `ICompositeAddress` interface to pass the user address and name.
This means that we can update the component properties to only add the `user: ICompositeAddress` property and trigger the onClick property whenever the user clicks on the wallet component